Kitodo/src/main/java/org/kitodo/production/helper/tasks/HierarchyMigrationTask.java

```diff
@@ -0,0 +1,358 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.production.helper.tasks;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kitodo.api.MetadataEntry;
+import org.kitodo.api.dataformat.IncludedStructuralElement;
+import org.kitodo.api.dataformat.Workpiece;
+import org.kitodo.api.dataformat.mets.LinkedMetsResource;
+import org.kitodo.data.database.beans.Process;
+import org.kitodo.data.database.beans.Project;
+import org.kitodo.data.database.exceptions.DAOException;
+import org.kitodo.data.exceptions.DataException;
+import org.kitodo.exceptions.ProcessGenerationException;
+import org.kitodo.production.helper.Helper;
+import org.kitodo.production.metadata.MetadataEditor;
+import org.kitodo.production.process.ProcessGenerator;
+import org.kitodo.production.services.ServiceManager;
+import org.kitodo.production.services.data.ProcessService;
+import org.kitodo.production.services.dataeditor.DataEditorService;
+import org.kitodo.production.services.dataformat.MetsService;
+import org.kitodo.production.services.file.FileService;
+
+public class HierarchyMigrationTask extends EmptyTask {
+    private static final Logger logger = LogManager.getLogger(HierarchyMigrationTask.class);
+
+    /**
+     * Service that contains the meta-data editor.
+     */
+    private static final DataEditorService dataEditorService = ServiceManager.getDataEditorService();
+
+    /**
+     * Service to access files on the storage.
+     */
+    private static final FileService fileService = ServiceManager.getFileService();
+
+    /**
+     * Service to read and write METS file format.
+     */
+    private static final MetsService metsService = ServiceManager.getMetsService();
+
+    /**
+     * Service to generate processes.
+     */
+    private final ProcessGenerator processGenerator = new ProcessGenerator();
+
+    /**
+     * Service to read and write Process objects in the database or search
+     * engine index.
+     */
+    private static final ProcessService processService = ServiceManager.getProcessService();
+
+    /**
+     * Parent processes have already been created. Key is the identifier, the
```

Comment:
Parent Processes have already been created? Really? When?

Reply:
If they were added to this map, they were created beforehand. It happens in line 169, which calls `createParentProcess()` itself.